Tests/CSRFTests/CSRFTests.swift

```diff
@@ -17,99 +15,264 @@ class CSRFTests: XCTestCase {
         super.tearDown()
     }
     
-    func testThatDropletHasMiddlewareIsNotEmpty() throws {
-        drop = try setUpDroplet()
+    func testThatCSRFMiddlewareReturnsToken() throws {
+        var token: String!
         
-        XCTAssertTrue(!drop.middleware.isEmpty, "Droplet middleware should not be empty.")
-        guard let _ = drop.middleware.first as? CSRF else {
-            XCTFail("CSRF middleware should have been added.")
-            return
+        app = try makeApplication(withSession: true)
+        
+        let router = try app.make(Router.self)
```

Comment:
You create the `router` and `responde` a lot. Could you use the `setUp` and `tearDown` methods to avoid typing them over and over?